module/VuFind/src/VuFind/I18n/Translator/Loader/ExtendedIni.php

```diff
@@ -37,218 +38,209 @@
     }
 
     /**
-     * Search the path stack for language files and merge them together.
+     * Loads a file.
      *
-     * @param string $filename    Name of file to search path stack for.
-     * @param bool   $failOnError If true, throw an exception when file not found.
+     * @param string $path Absolute path to file.
      *
-     * @return TextDomain
+     * @return void
      */
-    protected function loadLanguageFile($filename, $failOnError = true)
+    protected function loadFile($path)
     {
-        // Don't load a file that has already been loaded:
-        if ($this->checkAndMarkLoadedFile($filename)) {
-            return new TextDomain();
+        if (in_array($path, $this->list)) {
+            throw new \RuntimeException(
+                "Circular chain of loaded language files."
+            );
         }
 
-        $data = false;
-        foreach ($this->pathStack as $path) {
-            if (file_exists($path . '/' . $filename)) {
-                // Load current file with parent data, if necessary:
-                $current = $this->loadParentData(
-                    $this->reader->getTextDomain($path . '/' . $filename)
-                );
-                if ($data === false) {
-                    $data = $current;
-                } else {
-                    $data->merge($current);
-                }
-            }
-        }
-        if ($data === false) {
-            // Should we throw an exception? If not, return an empty result:
-            if ($failOnError) {
-                throw new InvalidArgumentException(
-                    "Ini file '{$filename}' not found"
-                );
-            }
-            return new TextDomain();
+        $this->list[] = $path;
+
+        if (!isset($this->dict[$path])) {
+            $this->readFile($path);
         }
 
-        return $data;
+        if ($extend = $this->dict[$path]['extend']) {
+            $this->loadFile($extend);
+        }
     }
 
     /**
-     * Support method for loadLanguageFile: retrieve parent data.
+     * Reads a language file.
      *
-     * @param TextDomain $data TextDomain to populate with parent information.
+     * @param string $path Absolute path to file.
      *
-     * @return TextDomain
+     * @return void
      */
-    protected function loadParentData($data)
+    protected function readFile($path)
     {
-        if (!isset($data['@parent_ini'])) {
-            return $data;
+        $data = ($exists = is_file($path))
+            ? $this->reader->getTextDomain($path) : new TextDomain();
+
+        if ($fallback = $data[self::FALLBACK] ?? null) {
+            $this->fallback = $this->fallback ?? $fallback;
+            unset($data[self::FALLBACK]);
         }
-        $parent = $this->loadLanguageFile($data['@parent_ini']);
-        $parent->merge($data);
-        return $parent;
+
+        if ($extend = $data[self::EXTEND] ?? null) {
+            $dir = $extend[0] === '/' ? APPLICATION_PATH : dirname($path);
```

Comment:
Absolute paths will be treated relative (appended) to `APPLICATION_PATH`.

Reply:
Is there a particular reason for this? I would expect an absolute path to actually be absolute, but maybe I'm misunderstanding the context.

Reply:
Ahh ... the was mainly motivated by the possibility to specify parent configuration files relative to `APPLICATION_PATH`.

But I see your point and I agree particularly as I can imagine scenarios where language files actually reside outside of `APPLICATION_PATH`.

I will change this to always resolve relative to the directory of the file containing the directive in order to avoid confusion.

Reply:
Sounds good. If the use case for config files relative to `APPLICATION_PATH` is strong, it might make sense to introduce some token that expands to the value of the constant... but I wouldn't want to overcomplicate things unless it would really be valuable.